qiskit/transpiler/preset_passmanagers/level0.py

```diff
@@ -171,7 +172,7 @@ def _swap_condition(property_set):
                 plugin_config=unitary_synthesis_plugin_config,
             ),
             UnrollCustomDefinitions(sel, basis_gates),
-            BasisTranslator(sel, basis_gates),
+            BasisTranslator(sel, basis_gates, target),
```

Comment:
My thinking was to pass the `Target` directly to each pass and have the pass extract the information it needs from the target (and to ensure the target class has helper methods to do that access in a performance conscience way). So in a world where the minimum supported backend version in terra was >=2 we can just make each pass in the preset pass managers only take a target object.

Reply:
That's fair, and definitely appreciate the performance benefits of keeping access to the convenience methods. When more passes make use of `Target`, and when it's expanded in scope a bit (via mixins or extensions), it may make sense to revisit this to maintain a separation of concerns between the passes.